hbase-common/src/main/java/org/apache/hadoop/hbase/util/ByteBufferUtils.java

```diff
@@ -778,6 +778,30 @@ public static int findCommonPrefix(ByteBuffer left, int leftOffset, int leftLeng
     return result;
   }
 
+  /**
+   * Find length of common prefix in two arrays.
+   * @param left        ByteBuffer to be compared.
+   * @param leftOffset  Offset in left ByteBuffer.
+   * @param leftLength  Length of left ByteBuffer.
+   * @param right       Array to be compared
+   * @param rightOffset Offset in right Array.
+   * @param rightLength Length of right Array.
+   */
+  public static int findCommonPrefix(ByteBuffer left, int leftOffset, int leftLength, byte[] right,
+    int rightOffset, int rightLength) {
+    int length = Math.min(leftLength, rightLength);
+    int result = 0;
+
+    while (
```

Comment:
This could also be improved by comparing 8 bytes each time.

You can see the code in ByteBufferUtils.compareToUnsafe.

Anyway, can be a follow on issue.

Reply:
Nice catch! Seems like it's a bit out of the scope of this PR like you mentioned. I'll create a follow up JIRA for optimizing the `ByteBufferUtils.findCommonPrefix` methods to compare 8 bytes at a time (instead of just a single byte at a time).

Reply:
I created this JIRA for the follow-up task: [HBASE-28025](https://issues.apache.org/jira/browse/HBASE-28025)